### PR TITLE
Cleaning of irf/io.py to introduce load_CTA_IRF helper function 

### PR DESCRIPTION
**Pull request by registerrier**
This PR is a first step in cleaning up suggested in issue #1551.

It removes all CTAPerf and associated tables in irf/io.py which are connected to sensitivity estimation with prod 2 CTA IRFs. Those will be moved in gammapy-extra.

The `load_CTA_IRF` helper function uses a default location for the CTA prod 3 IRF file. This might not be a good idea. 
the simulate_3d.ipynb notebook is modified to use this helper function .

IS this OK @jjlk ?

### CODE REVIEW

File: tutorials/simulate_3d.ipynb [hunk 1]
```diff
@@ -151,9 +188,20 @@
...
...IwHcwswvjQnM/Alm/l/Acrb8V+EFKVcR0TfpBlJliOgnqHlBzPuI6AVE9EQ0K3R/l5oXtNyaiP6cmvd6xxevXBXKv730JojoUUT0ntDXK+POnUT0cSJ6XmjzaiJ6QEi/MzUvprmKiH6TiD5BRHcKeXG76BcA+NYwzudoDYCI/pCIHhHOn0FEf09EfwHgm0WZOxPRG4joinAs8yZMGBoTKUwYAqcDuCqR/xkAj2bmBwN4MoBfLy1DRI9DszfOw5j56wD8IjO/Hs0Wyt/LzA9i5uX++ER0ZwC/BeAJofyTSm6AiE5E8+KXJzPzv0azg/APiSLXh7H9BoD/GtJ+Bs1eUA8G8CYA9zKaPg/AX4VxvijR/90APA8NGTwazfsSIn4NwIu42ZL8CQBeXnJPEyZ0wbR19oTBQUQvQbNn1C1BkB0P4MXUvHJxF8DXGNW8Mv8WwG8z85cAgJlzL7p5OBoz1scKy0fcH8DHmPnvw/WFAJ4F4H+G67g77rsR3mMQ7vHfh37+mIg+X9iXhYehebPbPwEAEV2M9ndwWmOlAwDcjoi+kpt3DEyYMCgmUpgwBK5BM4MFADDzs4IZ5cqQ9BwAnwbwdWi00y8bbXhlCHV74deWl/VSuDl87mLvf5OrY2GOtoZ+ojj3xj0D8I1SI5owYSxM5qMJQ+AyACcSkTS3fIU4vz2A65h5AeAcNK/H1PDK/AmA7yeirwCaF82H9BvRvDNb4x0Avj1s5y3L5xBfaXm/cH0OgL/I1PlrAP8h9PMYAHc0yuhxfhzNC2dmRHRP7L1H4HIAjyCik6h5f4c0e/0JgGfHi6BNTZgwCiZSmNAb3Gy1+11ohPHHiOhdaMwvPxmK/G8ATyOid6IxidxkNGOWYeY/RrPn/5XUvGUs2vN/B8BLo6NZjOWfADwTwBuJ6H3Ye3WjxqOI6Gg80Lxm9RkAXkdEV6N5n/NLnboRzwPwGCK6CsDj0GxFrk067wcwD07v5wD4GzQvjrkawC8j+GK4eefBBWhI7U/R9tH8yP9r7w5OEAaCKID+sRCryMWCrMkeLMhiJLIeNkwBCUGE9yrY22eZYX6SpWZ15SuzLhVO4XQ27LRtJ33GGGtV3ZI8xmxCg79lpgD7XZM8q+qS5J3k/uP3wGF+CgA0MwUAmlAAoAkFAJpQAKAJBQDaF2US4uTY1cIoAAAAAElFTkSuQmCC\n",
```
[the remark this answers is not in the document]
Comment (registerrier) [reply tutorials/simulate_3d.ipynb]: Done.

**Comment by cdeil** [comment]
@registerrier - OK.

> Simulations should be performed on reduced datasets. So that the most natural approach to me is extract the reduced IRFs from a (set of) fake empty observations.

You could try and see if this "just works" with https://docs.gammapy.org/0.8/api/gammapy.data.ObservationCTA.html
passing it to MapMaker.
(well, there's always some little changes needed, but I think in principle it should work)

We're not sure if we want to keep it as a second observation class, but the use case to build one from scratch and use it for simulation remains in whatever form this code takes, so 👍 to move in this direction.

**Comment by registerrier** [comment]
I have marked `xfail` the `test_sensivity` in gammapy/spectrum which is using CTAPerf.

The plan is to modify `SensitivityEstimator` to use `SpectrumSimulation`. This requires a number of changes, so this will go in a later PR.

**Comment by registerrier** [comment]
There is a Travis CI fail here: https://travis-ci.org/gammapy/gammapy/jobs/452289084#L4201

It is apparently due to the impossibility to find the gammapy dataset. Note that this test is done without gammapy-extra. I thought that using data stored in $GAMMAPY_DATA would work even without gammapy-extra. Am I wrong? @Bultako @cdeil

**Comment by Bultako** [comment]
@registerrier 
The config file for Travis declares the path of `GAMMAPY_DATA` in the following way.

https://github.com/gammapy/gammapy/blob/52b2d07786376c31610531251a7aa9f61e324401/.travis.yml#L167-L169

This is something that has to be changed and use `gammapy download datasets` instead. But as it is now, `GAMMAPY_DATA` relies on `GAMMAPY_EXTRA` in Travis CI.

There's some discussion about removing  `GAMMAPY_EXTRA` env var in https://github.com/gammapy/gammapy/issues/1889 and use only `GAMMAPY_DATA`

Basically `GAMMAPY_EXTRA` = `GAMMAPY_DATA` + `test_datasets/` 
After some thinking, I would be in favor of adding a flag to `gammapy download datasets` to also download `test_datasets/`  and replace in the code all references to `GAMMAPY_EXTRA` by `GAMMAPY_DATA`.

In the mean time, consider that Travis tests explicitly not requiring `GAMMAPY_EXTRA` will fail when working with data files.

**Comment by cdeil** [comment]
We still have a ton of tests that access GAMMAPY_EXTRA:
```
$ ack GAMMAPY_EXTRA gammapy | wc -l
106
```
@registerrier - Please do that for this PR.

@Bultako - I think getting rid of GAMMAPY_EXTRA for Gammapy testing isn't so simple. Probably adding `gammapy download data --tests` or something similar is the way to go. But we would first need to think about and be clear if test data should only accumulate forever, or if some test data files are allowed to change or be removed, in which case a version-coupling of code and tests data is needed, which we don't have for `gammapy download data`. Maybe the whole decision to not have version-coupling for `gammapy download data` to avoid unnecessary duplicated copies for multiple versions users have wasn't so great (still changing, will want to remove something in the coming years). To be discussed ...

File: gammapy/irf/io.py [hunk 2]
```diff
@@ -1,464 +1,41 @@
...
     The IRF format should be compliant with the one discussed
     at http://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/.
-    Waiting for a new public production of the CTA IRF,
-    we'll fix the missing pieces.
 
-    This class is similar to `~gammapy.data.DataStoreObservation`,
-    but only contains IRFs (no event data or livetime info).
-    TODO: maybe re-factor code somehow to avoid code duplication.
+    The various IRFs are accessible with the following keys
+    'aeff' is a `~gammapy.irf.EffectiveAreaTable2D`
```
Comment (cdeil) [inline gammapy/irf/io.py]: This will look much better if you do a bullet list like this (empty line is needed for RST):
```
The various IRFs are accessible with the following keys:

- 'aeff' is a `~gammapy.irf.EffectiveAreaTable2D`
- ...
```
Otherwise it all appears as one wrapped paragraph in the HTML and is harder to read.

**Comment by registerrier** [comment]
Most issues solved. Merge now.
The cta_sensitivity.ipynb notebook fails now because CTAPerf is gone. Some adaptation to `SensitivityEstimator` is required to solve this.  This will be done in a later PR.
